### PR TITLE
Pins endpoint

### PR DESCRIPTION
**Pull request by poom201211**
Wrote endpoint for pins including:
pin/?type=all
pin/?type=restaurant
pin/?type=cafe
pin/?type=attraction

### CODE REVIEW

File: api/controllers/pinController.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+exports.types = (req, response) => {
+  var type = req.query.type;
+
+  const typeList = ["cafe", "attraction", "restaurant", "all"];
+
+  if (!typeList.includes(type)) {
+    return response.status(500).json({ message: "Invalid type" });
+  }
+  if (type === "all") {
+    pin.find({}, (err, pin) => {
```
Comment (HAYATO-Kz) [inline api/controllers/pinController.js:13]: Always handle error

File: api/controllers/pinController.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      response.status(200).json(pin);
```
Comment (HAYATO-Kz) [inline api/controllers/pinController.js]: Don't forget to use return

File: api/controllers/pinController.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    });
+  } else {
+    pin.find({ type: req.query.type }, (err, pin) => {
```
Comment (HAYATO-Kz) [inline api/controllers/pinController.js:22]: Always handle error

File: api/controllers/pinController.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      response.status(200).json(pin);
```
Comment (HAYATO-Kz) [inline api/controllers/pinController.js]: Don't forget to use return

**Review by HAYATO-Kz — DISMISSED** [review]

File: api/controllers/pinController.js (hunk 1 above)
Comment (poom201211) [reply api/controllers/pinController.js:13]: fixed

File: api/controllers/pinController.js (hunk 2 above)
Comment (poom201211) [reply api/controllers/pinController.js]: fixed

File: api/controllers/pinController.js (hunk 3 above)
Comment (poom201211) [reply api/controllers/pinController.js:22]: fixed

File: api/controllers/pinController.js (hunk 4 above)
Comment (poom201211) [reply api/controllers/pinController.js]: fixed

**Review by HAYATO-Kz — APPROVED** [review]

**Pull request merged**